app/controllers/songs_controller.rb

```diff
@@ -65,48 +66,52 @@ def adjust_lang_params
 
   def set_songs_to_check
     @songs_to_check ||= {}
-    @songs_to_check[:changed] = Song.recently_changed.map { |s|
-      {
-        title: [s.custom_title, s.firstline_title, s.chorus_title].reject(&:blank?).first,
-        model: s,
-        edit_timestamp: s.updated_at
-      }
-    }
+    @songs_to_check[:changed] = sort_songs(Song.recently_changed.map { |song| admin_song_entry(song.titles.values.first, song) })
+    @songs_to_check[:duplicates] = sort_songs(Song.duplicates.map { |song| admin_song_entry(song.titles.values.first, song) }) if super_admin
 
-    non_duplicate_title_ids = Song.select("MIN(id) as id").group(:firstline_title).collect(&:id)
-    duplicate_titles = Song.select(:firstline_title).where.not(id: non_duplicate_title_ids)
-    @songs_to_check[:duplicates] = Song.where(firstline_title: duplicate_titles).map { |s|
-      {
-        title: [s.custom_title, s.firstline_title, s.chorus_title].reject(&:blank?).first,
-        model: s,
-        edit_timestamp: s.updated_at
-      }
-    }
   end
 
   def set_song
     @song = Song.find(params[:id] || params[:s])
   end
 
-  def set_songs
+  def set_songs(admin: false)
     @songs = []
     Song.all.includes(books: :song_books).each do |song|
-      song.titles.each do |t|
-        @songs << {
-          title: t[1],
-          model: song,
-          books: song.song_books.map {|sb| [sb.book.name, sb.index] }.to_h,
-          edit_timestamp: song.updated_at
-        }
+      song.titles.values.each do |title|
+        @songs << (admin ? admin_song_entry(title, song) : song_entry(title, song))
       end
     end
-    @songs.sort_by! { |s| clean_for_sorting(s[:title]) }
+    sort_songs(@songs)
+  end
+
+  def admin_song_entry(title, song)
+    {
+      title: title,
+      id: song.id,
+      books: song.book_indices,
+      edit_timestamp: song.updated_at,
+      last_editor: song.last_editor
+    }
+  end
+
+  def song_entry(title, song)
+    {
+      id: song.id,
+      title: title,
```

Comment:
Do you think long-term it would be good to put chorus_title in here rather than duplicating the song twice? Makes the sorting a bit harder I guess. Maybe some other field like `redirect_to: song.id` so choruses can just point to the original song?
Anyway, not important for this PR, just thinking...

Reply:
yea, multiplying records for each title does bulk things up a bit, but yea much easier for sorting.

An alternative could be a list of unique songs by id, and a separate list mapping titles to ids.